cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1,24 +1,269 @@
+                        {}
+                        """
+                    "#, value(
+                    customization
+                        .profile_bash
+                        .as_deref()
+                        .map(|profile_bash| indent::indent_all_by(2, profile_bash))
+                        .unwrap_or("".to_string()),
+                    )}),
+                );
+                profile_table.insert(
+                    "zsh",
+                    value(formatdoc! {r#"
+                        """
+                        {}
+                        """
+                    "#, value(
+                    customization
+                        .profile_zsh
+                        .as_deref()
+                        .map(|profile_zsh| indent::indent_all_by(2, profile_zsh))
+                        .unwrap_or("".to_string()),
+                    )}),
+                );
+            },
+        };
+
+        if let Some(ref _custom_profile) = customization.profile_common {
+            // .map(|table| table.insert("common", indent::indent_all_by(2, custom_profile)))
+            // manifest[MANIFEST_PROFILE_TABLE_KEY];
+            dbg!(MANIFEST_PROFILE_TABLE_KEY);
+            todo!();
+        }
+
+        profile_table.decor_mut().set_prefix(indoc! {r#"
+            # Scripts defined in the `[profile]` section are *sourced* by *your shell* and
+            # inherit environment variables set in the `[vars]` section and by `[hook]` scripts.
+            # The `profile.common` script is sourced by all shells and special care should be
+            # taken to ensure compatibility with all shells. The `profile.bash` and `profile.zsh`
+            # scripts are then sourced by the corresponding shell.
+        "#});
+
+        manifest.insert(MANIFEST_PROFILE_TABLE_KEY, Item::Table(profile_table));
+
+        // `[options]` table
+        let mut options_table = Table::new();
+
+        options_table.decor_mut().set_prefix(indoc! {r#"
+            # Additional options can be set in the `[options]` section. Refer to
+            # manifest.toml(1) for a list of available options.
+        "#});
+
+        // `systems` array
+        options_table.insert_formatted(
+            &Key::new(MANIFEST_SYSTEMS_ARRAY_KEY).with_leaf_decor(Decor::new(
+                indoc! {r#"
+
+                    # An environment that works on one system is guaranteed to work on the same type
+                    # of system, but other systems may not have the same packages available, etc.
+                    # In order to use the environment on a system you must explicitly add it to the
+                    # `options.systems` list.
+                "#},
+                " ",
+            )),
+            value(Array::from_iter(systems)),
+        );
+
+        manifest.insert_formatted(
+            &Key::new(MANIFEST_OPTIONS_TABLE_KEY),
+            Item::Table(options_table),
+        );
+
+        RawManifest(manifest)
```

Comment:
quib: I'm still thinking whether we need a `check` method and whether that would be on `RawManifest` or `TypedManifest`
I do think we need to try parse what we have created here (at least in a test!).

Reply:
> [!IMPORTANT]
> suggestion: this method needs to generate both "`v0`" (no `version` field) and `v1` (`version = 1`) manifests
> note that @zmitchell is currently also working on making `flox init` compatible with `v1` manifests there is some ovelap in both your work

Reply:
I think that check should be done when converting/parsing to `TypedManifest` i.e. parse don't validate

Reply:
I'm unsure what to do about this `check` feature but I'll add unit tests to cover `RawManifest::new_documented`.

Reply:
I think that's all we need now